lib/src/api.dart

```diff
@@ -0,0 +1,118 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:http/http.dart' as http;
+
+import 'exceptions.dart';
+import 'weather_api_openmeteo_sdk_generated.dart';
+
+mixin Parameter<Api extends BaseApi, Time extends TimeType> on Enum {
+  Variable get variable;
+  Aggregation get aggregation => Aggregation.none;
+  int get altitude => 0;
+  int get pressureLevel => 0;
+  int get depth => 0;
+  int get depthTo => 0;
+
+  int get hash => computeHash(
+        variable: variable,
+        altitude: altitude,
+        aggregation: aggregation,
+        pressureLevel: pressureLevel,
+        depth: depth,
+        depthTo: depthTo,
+      );
+}
+
+int computeHash({
+  required Variable variable,
+  required Aggregation aggregation,
+  required int altitude,
+  required int pressureLevel,
+  required int depth,
+  required int depthTo,
+}) =>
+    Object.hash(variable.value, aggregation.value, altitude, pressureLevel,
+        depth, depthTo);
+
+Map<int, ApiParameter> makeHashes<ApiParameter extends Parameter>(
+        List<ApiParameter> values) =>
+    {for (final val in values) val.hash: val};
+
+sealed class TimeType {}
```

Comment:
I quite don't get these empty classes, can you elaborate?
Is this for naming?

Reply:
The empty classes are used by the weather parameter enums (e.g. `WeatherHourly`) to determine where they are permitted to go in `Response`. They only exist as generic parameters.

Tl;dr they stop you from accessing the responses Daily variables with an Hourly key, generating compiler warnings.

Reply:
The result is that something like:
```dart
final response = await api.request(...);
final data = response.currentData[WeatherHourly.temperature_2m];
```
... throws a warning, because `WeatherHourly` is the wrong key type and will not resolve correctly.